AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -560,6 +560,28 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)__setNeedsLayout
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (!_flags.isMeasured) {
+    return;
```

Comment:
Is this really the right behavior?  If I call -setNeedsLayout on an un-measured node, it should probably result in a valid layout, even if it must do the measurement on the main thread.  I'm not confident about this, interested in hearing your considerations.